alf/examples/main.py

```diff
@@ -11,56 +11,81 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Off-policy training using ActorCriticAlgorithm.
+r"""Train using ActorCriticAlgorithm.
+
 To run actor_critic on gym CartPole:
 ```bash
-python actor_critic.py \
-  --root_dir=~/tmp/cart_pole_ppo \
-  --alsologtostderr \
-  --gin_file=ppo_cart_pole.gin
+python main.py \
+  --root_dir=~/tmp/cart_pole \
+  --gin_file=ac_cart_pole.gin \
+  --gin_param='create_environment.num_parallel_environments=8' \
+  --alsologtostderr
+```
+
+You can visualize playing of the trained model by running:
+```bash
+python main.py \
+  --root_dir=~/tmp/cart_pole \
+  --gin_file=ac_cart_pole.gin \
+  --play \
+  --alsologtostderr
 ```
+
 """
 
+import os
+
 from absl import app
 from absl import flags
 from absl import logging
 
-import gin
-import tensorflow as tf
+import gin.tf.external_configurables
 
-from alf.examples.actor_critic import create_algorithm, create_environment
-from alf.algorithms.on_policy_algorithm import OffPolicyAdapter
-from alf.trainers import off_policy_trainer, on_policy_trainer
-import alf.utils.external_configurables
+from alf.algorithms.actor_critic_algorithm import create_ac_algorithm
+from alf.environments.utils import create_environment
+from alf.trainers import on_policy_trainer, off_policy_trainer
+from alf.algorithms.off_policy_algorithm import OffPolicyAlgorithm
 from alf.utils import common
-import alf.algorithms.ppo_loss
+import alf.utils.external_configurables
+
+flags.DEFINE_string('root_dir', os.getenv('TEST_UNDECLARED_OUTPUTS_DIR'),
+                    'Root directory for writing logs/summaries/checkpoints.')
+flags.DEFINE_multi_string('gin_file', None, 'Paths to the gin-config files.')
+flags.DEFINE_multi_string('gin_param', None, 'Gin binding parameters.')
+flags.DEFINE_bool('play', False, 'Visualize the playing')
 
 FLAGS = flags.FLAGS
 
 
 @gin.configurable
-def train_eval(train_dir, evaluate=True, debug_summaries=False):
-    """A simple train and eval for ActorCriticAlgorithm."""
+def train_eval(train_dir,
+               algorithm_ctor=create_ac_algorithm,
+               evaluate=True,
+               debug_summaries=False):
```

Comment:
comment this function. And also note that play.algorithm_ctor is bind with this algorithm_ctor, as it might cause confusion